hack/verify-external-dependencies-version.sh

```diff
@@ -26,6 +26,19 @@ source "${KUBE_ROOT}/hack/lib/init.sh"
 
 kube::golang::verify_go_version
 
-cd "${KUBE_ROOT}"
+# Ensure that we find the binaries we build before anything else.
+export GOBIN="${KUBE_OUTPUT_BINPATH}"
+PATH="${GOBIN}:${PATH}"
 
-go run cmd/verifydependencies/verifydependencies.go "${KUBE_ROOT}"/build/dependencies.yaml
+# Install zeitgeist
+cd "${KUBE_ROOT}/hack/tools"
+GO111MODULE=on go install sigs.k8s.io/zeitgeist
+cd -
+
+# Prefer full path for running zeitgeist
+ZEITGEIST_BIN="$(which zeitgeist)"
+
+"${ZEITGEIST_BIN}" validate \
+  --local \
+  --base-path "${KUBE_ROOT}" \
+  --config "${KUBE_ROOT}"/build/dependencies.yaml
```

Comment:
since the dependencies.yaml now depends on an external tool, is it worth commenting in that file with a link to the tool and the format? is the external tool treatment of that file format stable, or is it still evolving? if the latter, do we depend on a particular version (if so, note that in the dependencies.yaml as well)

Reply:
@liggitt -- Added this in https://github.com/kubernetes/kubernetes/pull/98845/commits/b14accb3d4997fbb44cd212c1f4606ce147553c7:

```yaml
  # zeitgeist (https://github.com/kubernetes-sigs/zeitgeist) was inspired by
  # (and now replaces) the cmd/verifydependencies tool to verify external
  # dependencies across the repo.
  #
  # The zeitgeist dependencies.yaml file format is intended to be
  # backwards-compatible with the original tooling.
  #
  # In instances where the file format may change across versions, this meta
  # dependency check exists to ensure we're pinned to a known good version.
  #
  # ref: https://github.com/kubernetes/kubernetes/pull/98845
  - name: "zeitgeist"
    version: "v0.1.1-0.20210222132743-e06e27751b7f"
    refPaths:
    - path: hack/tools/go.mod
      match: sigs.k8s.io/zeitgeist\ v(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?
```

Reply:
heh, so meta